doc/AG0019.md

```diff
@@ -0,0 +1,63 @@
+﻿# Prevent use of InternalsVisibleTo Attribute
+
+ID: AG0019
+
+Type: Code Smell
+
+## Summary
+
+https://agoda-com.github.io/standards-c-sharp/unit-testing/only-test-the-public-interface.html
+
+The `InternalsVisibleTo` attribute should not be used as it violates encapsulation principles and creates tight coupling between test code and implementation details.
+
+## Explanation
+
+Making internal members visible to test assemblies is a common but problematic practice that:
+
+- Violates encapsulation principles
+- Creates tight coupling between tests and implementation
+- Makes tests brittle and harder to maintain
+- Makes refactoring more difficult
+- Encourages testing implementation details instead of behavior
+
+You should only test the public surface of your classes. If you find yourself needing to test internal methods, consider:
+
+- Making the method public if it represents actual API surface
+- Refactoring the code to better expose the behavior through public methods
+- Rethinking the design to avoid the need to test internals
+
+### Don't ❌
```

Comment:
Can we unify? Either use the cute icons, or not - for every doc (13 and before don't seem to have it)?